src/index.js

```diff
@@ -116,8 +116,39 @@ const taggedTemplateVisitor = (path, state) => {
   path.replaceWith(replacementNode);
 };
 
+const throwIfAfterStyledComponentsPlugin = plugins => {
+  let styledComponentsPluginIndex = -1;
+  let thisPluginIndex = -1;
+  plugins.forEach(([plugin], index) => {
+    if (plugin.key === 'styled-components') {
+      styledComponentsPluginIndex = index;
+    } else if (plugin.pre === pre) {
+      thisPluginIndex = index;
+    }
+  });
+  if (
+    styledComponentsPluginIndex !== -1 &&
+    styledComponentsPluginIndex < thisPluginIndex
+  ) {
+    throw new Error(
+      '`babel-plugin-namespace-styled-components` must be defined before the ' +
+        '`styled-components` plugin'
+    );
+  }
+};
```

Comment:
I think we can simplify this down by using `find` instead of `forEach`. A side effect of that would be a slightly optimized search as we can abort as soon as we find this plugin and it is before the styled-components plugin.

E.g., (psuedo-code ish)

``` js
let styledComponentsPluginIndex = -1;
plugins.find(([plugin], index) => {
   if(plugin.key === 'styled-components') {
     styledComponentsPluginIndex = index;
     return false;
  }

  if(plugin.pre === pre) {
    if(styledComponentsPluginIndex >= 0) {
     throw new Error('...');
   }
   
   return true;
  }
});
```

Although, seeing that written out, your original may be slightly more readable. It's a toss up, so either way.

Reply:
Sure, using `find` makes sense: 4a6a656